portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -5665,7 +5665,12 @@ public String getUserName(
 
 		if (user != null) {
 			if (userAttribute.equals(UserAttributes.USER_NAME_FULL)) {
-				userName = user.getFullName();
+				if ((user.getFullName() != null) ||
+					!user.getFullName(
+					).isEmpty()) {
+
+					userName = user.getFullName();
+				}
 			}
 			else {
 				userName = user.getScreenName();
```

Comment:
@MirnaGama Does this branch suffer from the same issue?

Reply:
It doesn't. 
When the method `getUserName(long userId, String defaultUserName) ` is called in `userName = getUserName(userId, userName)`, it explicitly sets the userAttribute value to _**USER_NAME_FULL**_.